applications/DamApplication/custom_constitutive/thermal_nonlocal_damage_3D_law.cpp

```diff
@@ -81,9 +79,11 @@ void ThermalNonlocalDamage3DLaw::CalculateMaterialResponseCauchy (Parameters& rV
   ElasticVariables.SetShapeFunctionsValues(rValues.GetShapeFunctionsValues());
   ElasticVariables.SetElementGeometry(rValues.GetElementGeometry());
   ElasticVariables.LameMu = 1.0+PoissonCoefficient;
-  ElasticVariables.ThermalExpansionCoefficient = MaterialProperties[THERMAL_EXPANSION]; 
-  ElasticVariables.ReferenceTemperature = rValues.GetProcessInfo()[REFERENCE_TEMPERATURE];
-    
+  ElasticVariables.ThermalExpansionCoefficient = MaterialProperties[THERMAL_EXPANSION];
+  /* Calculate Nodal Reference Temperature */
+  double NodalReferenceTemperature;
+  this->CalculateNodalReferenceTemperature(ElasticVariables,NodalReferenceTemperature);
+  
```

Comment:
Thanks @djvicente for the job. 
I just had one doubt: Why don't you store the interpolated NODAL_REFERENCE_TEMPERATURE in ElasticVariables.ReferenceTemperature? This way you won't need to modify the method CalculateThermalStrain...
If you wanted to leave that variable free just in case you use it in the future then it is fine. But if it is not the case, then we may avoid many modifications of this PR (because of the numerous calls to CalculateThermalStrain)

Reply:
Thank you @ipouplana for your comments.
I have been talking with @joaquinirazabal, and we both agree that it could be convenient maintaining the Nodal Reference Temperature as an independent variable. To date, this variable indeed remains constant. Nevertheless, in the future, we might find interesting that this variable could be modified due to the uncertainty associated to certain processes: construction stage of the dam, occurrence of new grouting actions, among others. Besides, all the references to the method 'CalculateThermalStrain' have already been modified. 
Another thing would be if this change affected to certain processes of other applications, for example, the Poro. If this is the case, tell me and I will change it again. What do you think?

Finally, I have one doubt related to your change: https://github.com/KratosMultiphysics/Kratos/pull/2129/commits/d304e8e797c29bd9282c7a4377caede586409ff3
It is necessary to include 'DamNodalReferenceTemperatureProcess::Pointer' in such class? I don't know at all how does it work that reference to '::Pointer'. If it is compulsory, maybe I have to change this issue for the rest of the classes of the DamApplication. Could you guide me in this regard? 

Thanks!

Reply:
Perfect @ipouplana. I didn't update your last changes and so, I just saw the change in the last process. Now, I can see that it has been changed in all the processes, so it makes sense. 
Thanks!